org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/MethodBinding.java

```diff
@@ -244,6 +244,9 @@ protected IAnnotationBinding[] filterTypeAnnotations(org.eclipse.jdt.internal.co
 			final boolean isConstructor = this.isConstructor();
 			for (int i = 0; i < length; i++) {
 				org.eclipse.jdt.internal.compiler.lookup.AnnotationBinding internalAnnotation = internalAnnotations[i];
+				if(internalAnnotation == null) {
```

Comment:
The internalAnnotations variable is returning null at this point. Additionally, similar handling already exists in the filterTypeAnnotations method for annotationInstance. This is the main reason I made the modification within the DOM itself. Could you please advise whether it would be better to handle this in the DOM or the compiler?

Reply:
Ideally, we should not populate any array with null elements. If it's possible to exclude that null annotation from the array, we should try and do that. If that's not possible, or too difficult to avoid, then we can consider a simple null check later in the late stage.

Reply:
Okay, I will do accordingly